src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java

```diff
@@ -0,0 +1,279 @@
+                    }
+                });
+
+            });
+            //caseWorkerRoles data setting to case worker profile
+            caseWorkerProfile.setCaseWorkerRoles(caseWorkerRoles);
+
+            //caseworkerworkarea request mapping
+            cwrdProfileRequest.getWorkerWorkAreaRequests().forEach(caseWorkerWorkAreaRequest -> {
+                CaseWorkerWorkArea caseWorkerWorkArea =  new CaseWorkerWorkArea(userProfileCreationResponse.getIdamId(),
+                        caseWorkerWorkAreaRequest.getAreaOfWork(), caseWorkerWorkAreaRequest.getServiceCode());
+                caseWorkerWorkAreas.add(caseWorkerWorkArea);
+            });
+
+            //caseWorkerWorkAreas setting to case worker profile
+            caseWorkerProfile.setCaseWorkerWorkAreas(caseWorkerWorkAreas);
+        } else {
+            //Add failed request to list and save to the exception table
+            log.error("{}::Idam register user failed", loggingComponentName);
+
+        }
+        return caseWorkerProfile;
+    }
+
+    /**
+     *  Idam_UP call.
+     *
+     */
+    private ResponseEntity<Object> createUserProfileInIdamUP(CaseWorkersProfileCreationRequest cwrdProfileRequest) {
+
+        Response response = null;
+        Object clazz = null;
+        try {
+            response = userProfileFeignClient.createUserProfile(createUserProfileRequest(cwrdProfileRequest));
+            if (response.status() == 201 || response.status() == 409) {
+                clazz = UserProfileCreationResponse.class;
+            } else {
+                clazz = ErrorResponse.class;
+            }
+
+        } catch (FeignException ex) {
+            log.error("{}:: UserProfile api failed:: status code {}", loggingComponentName, ex.status());
+            clazz = ErrorResponse.class;
+        }
+        return JsonFeignResponseUtil.toResponseEntity(response, clazz);
+    }
+
+    /**
+     * creating user profile request.
+     *
+     * */
+    public UserProfileCreationRequest createUserProfileRequest(CaseWorkersProfileCreationRequest cwrdProfileRequest) {
+
+        Set<String> userRoles = cwrdProfileRequest.getIdamRoles() != null ? cwrdProfileRequest.getIdamRoles() :
+                new HashSet<String>();
+        userRoles.add("cwd-user");
+        Set<String> idamRoles = getUserRolesByRoleId(cwrdProfileRequest);
+        if (idamRoles.size() > 0) {
+            userRoles.addAll(idamRoles);
+        }
+        //Creating user profile request
+        return  new UserProfileCreationRequest(
+                cwrdProfileRequest.getEmailId(),
+                cwrdProfileRequest.getFirstName(),
+                cwrdProfileRequest.getLastName(),
+                LanguagePreference.EN,
+                UserCategory.CASEWORKER,
+                UserTypeRequest.INTERNAL,
+                userRoles,
+                false);
+    }
+
+    /**
+     *  get the roleTypes and userTypes.
+     */
+    public  void getRolesAndUserTypes() {
+
```

Comment:
these calls should be on repository class, so that any service can use it.

Reply:
that is by default methods in the repository  and that will use any one.

Reply:
I mean you can refer PRD code how it has done PrdEnumServiceImpl.findAllPrdEnums(), similarly you can implement  roleTypeRepository.findAll () manually